lib/handler/redirect.c

```diff
@@ -79,10 +87,50 @@ static void redirect_internally(h2o_redirect_handler_t *self, h2o_req_t *req, h2
     h2o_send_error_503(req, "Internal Server Error", "internal server error", 0);
 }
 
+static h2o_iovec_t join_list(h2o_mem_pool_t *pool, h2o_iovec_vector_t *list, h2o_iovec_t delimiter)
+{
+    h2o_iovec_vector_t joined = (h2o_iovec_vector_t){ 0 };
```

Comment:
How about using alloca?

```c
h2o_iovec_t *joined = alloca(sizeof(*joined) * list->size * 2 + 1);
```

Reply:
done in https://github.com/h2o/h2o/pull/2186/commits/7405d448a51c179ab5668139ff539e3edddfe1f8